main/src/library/Console.flix

```diff
@@ -114,14 +114,14 @@ namespace Console {
     pub def hex(hexCode: String, s: a): String with ToString[a] =
         let text = ToString.toString(s);
         let hexOpt: Option[String] = {
-            use Option.flatMap;
-            let* hex = String.stripPrefix(substr = "#", hexCode);
-            if (String.length(hex) == 6) {
-                let* r = String.sliceLeft(end = 2, hex) |> stringToHex;
-                let* g = String.slice(start = 2, end = 4, hex) |> stringToHex;
-                let* b = String.slice(start = 4, end = 6, hex) |> stringToHex;
-                Some(rgb((r, g, b), text, isFg = true))
-            } else Option.None
+            forM (
+                hex <- String.stripPrefix(substr = "#", hexCode);
+                r <- String.sliceLeft(end = 2, hex) |> stringToHex;
+                g <- String.slice(start = 2, end = 4, hex) |> stringToHex;
+                b <- String.slice(start = 4, end = 6, hex) |> stringToHex;
+                if String.length(hex) == 6
```

Comment:
I think this rewrite is wrong. You know check `length == 6` too late-- which means that the slices may not work as intended? In either case, I think the control-flow has been obscured.

Reply:
Possibly it cannot fail because Flix is pretty good with having functions that are total, but I still think the order is un-natural.

Reply:
Perhaps we want:

```
// in Option
def cond(b: Bool): Option[Unit] = if (b) Some(()) else None
```

then you can do:

```
forM(
  _ <- Option.cond(String.length(hex) == 6)
```

Reply:
Thoughts?

Reply:
The `Option.cond` function seems redundant as we already have support for guards in `forM` and if you're not operating on monads you can easily wrap something in a specific monad by just writing `if (b) Some(x) else None`. I think this particular problem is just a matter of reordering where the guard appears.